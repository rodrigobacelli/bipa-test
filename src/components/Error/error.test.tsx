import { describe, it, expect, vi } from 'vitest';

import { render } from '../../tests/test-utils';

import { Error } from './';

describe('components/Error', () => {
  it('should accept classNames', () => {
    const result = render(<Error className="error-className" />);

    expect(
      result.container.querySelector('.error-className')
    ).toBeInTheDocument();
  });

  it('should handle Button click', async () => {
    const onRetry = vi.fn();
    const result = render(<Error onRetry={onRetry} />);

    await result.getByRole('button').click();

    expect(onRetry).toHaveBeenCalledTimes(1);
  });

  it('should render the Error Component', () => {
    const result = render(<Error className="error-className" />);

    expect(result).toMatchSnapshot();
  });
});
