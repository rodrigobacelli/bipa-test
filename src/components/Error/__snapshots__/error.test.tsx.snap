// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/Error > should render the Error Component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-full w-full items-center justify-center error-className"
        id="no-content"
      >
        <p
          class="flex flex-col items-center gap-2"
        >
          <svg
            class="h-8 w-8"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          An error occurred. Please try again later.
          <button
            class="flex items-center gap-1 rounded-sm border px-2 py-1 text-sm enabled:cursor-pointer border-gray-900 text-gray-900 enabled:hover:bg-gray-900/20 enabled:focus:bg-gray-800/10 disabled:border-gray-500 disabled:text-gray-500"
          >
            Retry
          </button>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-full w-full items-center justify-center error-className"
      id="no-content"
    >
      <p
        class="flex flex-col items-center gap-2"
      >
        <svg
          class="h-8 w-8"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        An error occurred. Please try again later.
        <button
          class="flex items-center gap-1 rounded-sm border px-2 py-1 text-sm enabled:cursor-pointer border-gray-900 text-gray-900 enabled:hover:bg-gray-900/20 enabled:focus:bg-gray-800/10 disabled:border-gray-500 disabled:text-gray-500"
        >
          Retry
        </button>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
